saleor/checkout/utils.py

```diff
@@ -969,3 +969,10 @@ def get_or_create_checkout_metadata(checkout: "Checkout"):
         return checkout.metadata_storage
     else:
         return CheckoutMetadata.objects.create(checkout=checkout)
+
+
+def get_checkout_metadata(checkout: "Checkout"):
```

Comment:
# 

In my opinion, we should double-check all occurrences of `get_or_create_checkout_metadata` and replace them with `get_checkout_metadata`. If the issue has already occurred once, it's only a matter of time before `get_or_create_checkout_metadata` causes the same issue again. Using `get_or_create_checkout_metadata` in one thread can create problems with saving metadata in a second thread. If your checkout doesn't have `metadata_storage`, you create a dummy object and maybe call `.save()` on it. If another thread calls `get_or_create_checkout_metadata` in the meantime, we will get an exception that the object already exists..

Reply:
You are right. I've missed some places that should be modified. Fixed it.